scipy/stats/_stats_py.py

```diff
@@ -7868,9 +7878,8 @@ def ks_2samp(data1, data2, alternative='two-sided', method='auto'):
         success, d, prob = _attempt_exact_2kssamp(n1, n2, g, d, alternative)
         if not success:
             mode = 'asymp'
-            if original_mode == 'exact':
-                warnings.warn(f"ks_2samp: Exact calculation unsuccessful. "
-                              f"Switching to method={mode}.", RuntimeWarning)
+            warnings.warn(f"ks_2samp: Exact calculation unsuccessful. "
+                          f"Switching to method={mode}.", RuntimeWarning)
```

Comment:
Should probably fix the `stacklevel` while we're at it.
```suggestion
                          f"Switching to method={mode}.", RuntimeWarning,
                          stacklevel=3)
```